service/ui_plugin_build_baron.go

```diff
@@ -321,43 +386,270 @@ func (aes *altEndpointSuggest) parseResponse(r io.ReadCloser) ([]thirdparty.Jira
+func (aes *altEndpointSuggest) GetTimeout() time.Duration {
+	return time.Duration(aes.timeoutSecs) * time.Second
+}
 
-	url := aes.bbProj.AlternativeEndpointURL
-	url = strings.Replace(url, "{task_id}", t.Id, -1)
-	url = strings.Replace(url, "{execution}", strconv.Itoa(t.Execution), -1)
+/////////////////////////////
+// multiSourceSuggest type //
+/////////////////////////////
+
+type multiSourceSuggest struct {
+	jiraSuggester suggester
+	altSuggester  suggester
+}
 
-	req, err := http.NewRequest(http.MethodGet, url, nil)
+func (mss *multiSourceSuggest) Suggest(t *task.Task) ([]thirdparty.JiraTicket, string, error) {
+	var tickets []thirdparty.JiraTicket
+	var source string
+	var err error
+
+	if mss.altSuggester != nil {
+		tickets, source, err = mss.raceSuggest(t)
+	} else {
+		source = JIRASource
+		tickets, err = mss.jiraSuggester.Suggest(context.TODO(), t)
+	}
+	return tickets, source, err
+}
+
+// raceSuggest returns the JIRA ticket results from the altEndpoint suggester if it returns
+// within its configured interval, and returns the JIRA ticket results from the fallback suggester
+// otherwise.
+func (mss *multiSourceSuggest) raceSuggest(t *task.Task) ([]thirdparty.JiraTicket, string, error) {
+	type result struct {
+		Tickets []thirdparty.JiraTicket
+		Error   error
+	}
+
+	// thirdparty/jira.go and thirdparty/http.go do not expose an API that accepts a context.Context.
+	fallbackCtx := context.TODO()
+	fallbackChan := make(chan result, 1)
+	go func() {
+		suggestions, err := mss.jiraSuggester.Suggest(fallbackCtx, t)
+		fallbackChan <- result{suggestions, err}
+		close(fallbackChan)
+	}()
+
+	altEndpointTimeout := mss.altSuggester.GetTimeout()
+	altEndpointCtx, altEndpointCancel := context.WithTimeout(context.Background(), altEndpointTimeout)
+	defer altEndpointCancel()
+	suggestions, err := mss.altSuggester.Suggest(altEndpointCtx, t)
+
+	// If the alternative endpoint didn't respond quickly enough or didn't have results available,
+	// then we wait for the fallback results. Ideally we'd otherwise be able to cancel the request
+	// for fetching the fallback results, but we instead just return back to the caller without
+	// waiting for the associated goroutine to complete.
 	if err != nil {
-		return nil, err
+		grip.Warning(message.WrapError(err, message.Fields{
+			"message":   "failed to get results from alternative endpoint",
+			"task_id":   t.Id,
+			"execution": t.Execution,
+		}))
+
+		fallbackChanRes := <-fallbackChan
+		return fallbackChanRes.Tickets, JIRASource, fallbackChanRes.Error
 	}
 
-	if aes.bbProj.AlternativeEndpointUsername != "" {
-		req.SetBasicAuth(aes.bbProj.AlternativeEndpointUsername, aes.bbProj.AlternativeEndpointPassword)
+	return suggestions, BFSuggestionSource, nil
+}
+
+/////////////////////////////////
+// BF Suggestion Server Client //
+/////////////////////////////////
+
+func getBFSuggestionClient(bbProj evergreen.BuildBaronProject) (bfSuggestionClient, bool) {
+	var client bfSuggestionClient
```

Comment:
I think you can deleted this and just return `bbProf.bfSuggestionClient{}` on 465.